usaspending_api/data_load/fpds_loader.py

```diff
@@ -0,0 +1,314 @@
+
+                # Create recipient so we can link it later
+                columns, values, pairs = setup_load_lists(load_object, "legal_entity")
+                recipient_sql = "INSERT INTO legal_entity {} VALUES {} " "RETURNING legal_entity_id".format(
+                    columns, values, pairs
+                )
+                cursor.execute(recipient_sql)
+                results = cursor.fetchall()
+                load_object["transaction_normalized"]["recipient_id"] = results[0][0]
+                load_object["award"]["recipient_id"] = results[0][0]
+
+                # Create place of performance so we can link it later
+                columns, values, pairs = setup_load_lists(load_object, "place_of_performance_location")
+                pop_location_sql = "INSERT INTO references_location {} VALUES {} " "RETURNING location_id".format(
+                    columns, values, pairs
+                )
+                cursor.execute(pop_location_sql)
+                results = cursor.fetchall()
+                load_object["transaction_normalized"]["place_of_performance_id"] = results[0][0]
+                load_object["award"]["place_of_performance_id"] = results[0][0]
+
+                # Try to find an award for this transaction to belong to
+                find_matching_award_sql = "select id from awards where generated_unique_award_id = '{}'".format(
+                    load_object["generated_unique_award_id"]
+                )
+                cursor.execute(find_matching_award_sql)
+                results = cursor.fetchall()
+
+                # If there is an award, we still need to update it with values from its new latest transaction
+                if len(results) > 0:
+                    load_object["transaction_normalized"]["award_id"] = results[0][0]
+                    update_award_str = "place_of_performance_id = {}, recipient_id = {}".format(
+                        load_object["award"]["place_of_performance_id"], load_object["award"]["recipient_id"]
+                    )
+                    update_award_sql = "UPDATE awards SET {} where id = {}".format(update_award_str, results[0][0])
+                    cursor.execute(update_award_sql)
+                # If there is no award, we need to create one
+                else:
+                    columns, values, pairs = setup_load_lists(load_object, "award")
+                    generate_matching_award_sql = "INSERT INTO awards {} VALUES {} RETURNING id".format(columns, values)
+                    cursor.execute(generate_matching_award_sql)
+                    results = cursor.fetchall()
+                    load_object["transaction_normalized"]["award_id"] = results[0][0]
+
+                # Determine if we are making a new transaction, or updating an old one
+                find_matching_transaction_sql = (
+                    "select transaction_id from transaction_fpds "
+                    "where detached_award_proc_unique = '{}'".format(
```

Comment:
I don't know why, but collisions occur when we do it on `detached_award_procurement_id` but not on proc unique, even though in my discussions of the DB they seem to be functionally the same.

Reply:
Perhaps the conflict is being detected on commit? It's possible to have duplicates being loaded within a single transaction block and failing when the changes are being committed.

Reply:
Turns out the problem was that trying to update a unique column can collide with itself. I side-stepped this issue by just not having the proc_unique be modified on update (it should always be the same anyway)